chia/wallet/puzzles/load_clvm.py

```diff
@@ -86,36 +89,59 @@ def load_serialized_clvm(
     """
     hex_filename = f"{clvm_filename}.hex"
 
-    try:
-        if pkg_resources.resource_exists(package_or_requirement, clvm_filename):
-            # Establish whether the size is zero on entry
-            full_path = pathlib.Path(pkg_resources.resource_filename(package_or_requirement, clvm_filename))
-            output = full_path.parent / hex_filename
-            search_paths = [full_path.parent]
-            if include_standard_libraries:
-                # we can't get the dir, but we can get a file then get its parent.
-                chia_puzzles_path = pathlib.Path(pkg_resources.resource_filename(__name__, "__init__.py")).parent
-                search_paths.append(chia_puzzles_path)
-            compile_clvm(full_path, output, search_paths=search_paths)
-
-    except NotImplementedError:
-        # pyinstaller doesn't support `pkg_resources.resource_exists`
-        # so we just fall through to loading the hex clvm
-        pass
+    # Set the CHIA_DEV_COMPILE_CLVM_ON_IMPORT environment variable to anything except
+    # "" or "0" to trigger automatic recompilation of the Chialisp on load.
+    if recompile:
+        try:
+            if pkg_resources.resource_exists(package_or_requirement, clvm_filename):
+                # Establish whether the size is zero on entry
+                full_path = pathlib.Path(pkg_resources.resource_filename(package_or_requirement, clvm_filename))
+                output = full_path.parent / hex_filename
+                search_paths = [full_path.parent]
+                if include_standard_libraries:
+                    # we can't get the dir, but we can get a file then get its parent.
+                    chia_puzzles_path = pathlib.Path(pkg_resources.resource_filename(__name__, "__init__.py")).parent
+                    search_paths.append(chia_puzzles_path)
+                compile_clvm(full_path, output, search_paths=search_paths)
+
+        except NotImplementedError:
+            # pyinstaller doesn't support `pkg_resources.resource_exists`
+            # so we just fall through to loading the hex clvm
+            pass
 
     clvm_hex = pkg_resources.resource_string(package_or_requirement, hex_filename).decode("utf8")
     assert len(clvm_hex.strip()) != 0
     clvm_blob = bytes.fromhex(clvm_hex)
     return SerializedProgram.from_bytes(clvm_blob)
 
 
-def load_clvm(clvm_filename, package_or_requirement=__name__, include_standard_libraries: bool = False) -> Program:
+def load_clvm(
+    clvm_filename,
+    package_or_requirement=__name__,
+    include_standard_libraries: bool = False,
+    recompile: bool = True,
```

Comment:
is there a good reason to not just default this to?
```
recompile: bool = recompile_requested,
```

And not bother to introduce a new name `load_clvm_maybe_recompile()`. Are there still use cases for the function where we try to recompile unconditionally?

If so, I would still think that would be the uncommon case that should have a longer name.

Reply:
The concern was that other projects might be using it.  The fact it is buried down in our wallet puzzles directory seems both bad and an indication against that.  @trepca brought up the concern.

Reply:
although, come to think of it, the NFT wallet does de-compile and re-compile clvm code. I'm not sure why, but there is a dependency there too.

Reply:
It'd be great to separate into dev libraries, but currently even `chia-dev-tools` depends on `chia-blockchain`. For any serious coin development, developers need access to simulator which is in `chia-blockchain` too and all the standard puzzles.